### PR TITLE
fix User Counts updates on user add/remove

### PR DESCRIPTION
**Pull request by petre2dor**
Fix for #27607 
I refactored the existing code that update the counts on user enable/disable and added code to update the counts on user add/remove

Signed-off-by: Petre T <petre.tudor@dorkfarm.com>

### CODE REVIEW

**Comment by skjnldsv** [comment]
@petre2dor could you enable us to change this pr? :)

image

**Comment by petre2dor** [comment]
Hi @skjnldsv,
That checkbox is already checked.
image

**Comment by Pytal** [comment]
👆 attempt 2

**Comment by szaimen** [comment]
> I guess it would probably make sense to push this to a branch in this repo and invite Petre to the org for easier collaboration...

@skjnldsv would you be able to do so? :)

**Comment by petre2dor** [comment]
Hi everyone, 
Just a friendly reminder about this PR in the hope we can move this along.

Thanks :)

**Comment by szaimen** [comment]
@petre2dor thanks for the ping! I would be able to push your changes to a branch in this repo but I fear I am not able to invite you to the org which would mean that you would not be able to push to the new branch...

Apart from that, you can also fix the `Node` workflow by compiling the js locally and pushing it to the branch.

File: apps/settings/src/store/users.js [hunk 1]
```diff
@@ -143,22 +143,59 @@ const mutations = {
...
 		const user = state.users.find(user => user.id === userid)
 		user.enabled = enabled
-		// increment or not
-		if (state.userCount > 0) {
-			state.groups.find(group => group.id === 'disabled').usercount += enabled ? -1 : 1
-			state.userCount += enabled ? 1 : -1
-			user.groups.forEach(group => {
-				// Increment disabled count
-				state.groups.find(groupSearch => groupSearch.id === group).disabled += enabled ? -1 : 1
+		// this.commit('updateUserCounts', { user, actionType: enabled ? 'enable' : 'disable' })
```
Comment (artonge) [inline apps/settings/src/store/users.js:157]: I would prefer the `enable`/`disable` action type so the `update` one is not wrongly used in the future.

File: apps/settings/src/store/users.js [hunk 2]
```diff
@@ -143,22 +143,59 @@ const mutations = {
...
+		this.commit('updateUserCounts', { user, actionType: 'update' })
+	},
+	// update active/disabled counts, groups counts
+	updateUserCounts(state, { user, actionType }) {
+		const disabledGroup = state.groups.find(group => group.id === 'disabled')
+		switch (actionType) {
+		case 'update':
```
Comment (artonge) [inline apps/settings/src/store/users.js:164]: To not duplicate the code you can handle both `enable` and ` disable` like so:

```
		case 'enable':
		case 'disable':
			...
```

and keeping the current logic

**Review by artonge — CHANGES_REQUESTED** [review]

**Comment by skjnldsv** [comment]
> @skjnldsv would you be able to do so? :)

invited @petre2dor

**Comment by szaimen** [comment]
> invited @petre2dor

@petre2dor did you accept? then I could push your changes to a new branch...

**Comment by petre2dor** [comment]
> > invited @petre2dor
> 
> @petre2dor did you accept? then I could push your changes to a new branch...

Yes, I did today. Thanks!
I'll make my changes today or tomorrow.

**Comment by szaimen** [comment]
> Yes, I did today. Thanks!
> I'll make my changes today or tomorrow.

Shall I push your changes to the new branch in the meantime or do you want to make your new changes first and then push it to the new branch afterwards?

**Comment by petre2dor** [comment]
> > Yes, I did today. Thanks!
> > I'll make my changes today or tomorrow.
> 
> Shall I push your changes to the new branch in the meantime or do you want to make your new changes first and then push it to the new branch afterwards?

Please push my changes to a new branch first. 
I destroyed my dev machine and I need to rebuild it. No need to use my fork and switch later to Nexcloud repo.

**Comment by szaimen** [comment]
Superseded by https://github.com/nextcloud/server/pull/31697

**Comment by szaimen** [comment]
> Please push my changes to a new branch first.
> I destroyed my dev machine and I need to rebuild it. No need to use my fork and switch later to Nexcloud repo.

done with https://github.com/nextcloud/server/pull/31697

File: apps/settings/src/store/users.js (hunk 2 above)
Comment (petre2dor) [reply apps/settings/src/store/users.js:164]: thanks for the review @artonge. I made the chenges and pushed here https://github.com/nextcloud/server/pull/31697
please review it again

**Pull request closed without merging**